CONTRIBUTING.md

```diff
@@ -15,14 +15,16 @@ To hack on Pulumi, you'll need to get a development environment set up. You'll w
 - [pipenv](https://github.com/pypa/pipenv)
 - [Golangci-lint](https://github.com/golangci/golangci-lint)
 - [Yarn](https://yarnpkg.com/)
+- [Istanbul](https://github.com/gotwarlost/istanbul)
 
 ## Getting dependencies on macOS
 
-You can easily get all required dependencies with brew
+You can easily get all required dependencies with brew and npm
 
 ```bash
 brew install node pipenv python@3 typescript yarn go golangci/tap/golangci-lint
-brew cask install dotnet
+brew cask install dotnet dotnet-sdk
+npm install -g istanbul
```

Comment:
BTW, I ran into this recently on a machine that didn't have Istanbul. I was tempted to just install it globally like this, but ideally, we'd always use the version of it installed in the local `node_modules` as specified in the `package.json`.

https://github.com/pulumi/pulumi/blob/48a1d3c3cb624955df5f757866b41c5b0d662310/sdk/nodejs/package.json#L35

We do already update `PATH` in the `Makefile` to be able to find such tools inside `node_modules`...

https://github.com/pulumi/pulumi/blob/48a1d3c3cb624955df5f757866b41c5b0d662310/sdk/nodejs/Makefile#L13

So the calls to `istanbul` in the Makefile *should* use the one inside `./node_modules/.bin`...

https://github.com/pulumi/pulumi/blob/48a1d3c3cb624955df5f757866b41c5b0d662310/sdk/nodejs/Makefile#L46-L49

However, I couldn't get this to work on macOS... Despite us [setting](https://github.com/pulumi/pulumi/blob/48a1d3c3cb624955df5f757866b41c5b0d662310/build/common.mk#L93) `SHELL := /bin/bash` in `common.mk` that we include in the Makefile (which is apparently necessary on macOS), the calls to `istanbul` would not work because it could not be found. It turns out, it's due to a bug in the version of `make` on macOS. https://stackoverflow.com/a/21709821. I was able to get it working by installing a more recent version of `make` via `brew install make` and then setting `export PATH=/usr/local/opt/make/libexec/gnubin:$PATH` in my `~/.bash_profile` to use that newer version of `make`. Now when I run `make`, it'll use the version of istanbul that was installed in `./node_modules` after a `make ensure`.

Reply:
Ah, thanks for sharing your solution! Is this what we should be recommending to users?

Reply:
> Is this what we should be recommending to users?

Good question. Installing globally is likely the simplest explanation for contributors on macOS, so maybe just leave things as you've added it? Unless we can distill my notes above to more succinct instructions.

The only downside is potential mismatch between the version installed globally and the expected version referenced in the `package.json`.

Alternatively, we could consider changing these lines:

https://github.com/pulumi/pulumi/blob/48a1d3c3cb624955df5f757866b41c5b0d662310/sdk/nodejs/Makefile#L47-L49

to something like:

```
./node_modules/.bin/istanbul test --print none ./node_modules/.bin/_mocha -- --timeout 15000 'bin/tests/**/*.spec.js'
./node_modules/.bin/istanbul report text-summary
./node_modules/.bin/istanbul report text
```

Like we've done for some of the other calls:

https://github.com/pulumi/pulumi/blob/48a1d3c3cb624955df5f757866b41c5b0d662310/sdk/nodejs/Makefile#L20

https://github.com/pulumi/pulumi/blob/48a1d3c3cb624955df5f757866b41c5b0d662310/sdk/nodejs/Makefile#L17

@pgavlin, do you have a preference?

Reply:
On another note, does our tooling assume the use of bash? Will I run into trouble if I use something like zsh?

Reply:
> On another note, does our tooling assume the use of bash? Will I run into trouble if I use something like zsh?

Very possibly yes.

Reply:
> Alternatively, we could consider changing these lines:

If changing those lines works, I think that's a good idea.